_posts/2017-11-29-Express-Reloadable-Update.markdown

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "Optimizing Linked Package Development in Express.js Apps"
+date: 2017-11-29 14:18
+comments: true
+author: chris
+categories: [Express, Node, DX, @artsy/express-reloadable]
+---
+
+In [Modernizing Force](artsy.github.io/blog/2017/09/05/Modernizing-Force/) we discussed some of the tools we've been working with to modernize [Artsy.net](https://www.artsy.net/)'s development environment, from introducing Babel and React to the creation of [@artsy/stitch](https://github.com/artsy/stitch). Increasing overall development speed was another aim, and to that end we released [@artsy/express-reloadable](https://github.com/artsy/express-reloadable) which automatically hot-swaps Express.js code without the restart. In this post I'd like to cover some of the issues we've faced since then, and in particular our solution to library code-sharing in Express apps.
+
+<!-- more -->
+
+It's common to share NPM packages across projects, and oftentimes packages are developed in parallel. Package `A` depends on `B`, but `B` has a bug and you don't want to have to republish (and reinstall) the package in order to see changes made locally. `yarn link` (or `npm link`) was developed for instances like this and while it works great for stop and start processes where boot time is quick, it falls short if the development environment takes a while to load. In UI-rich environments like [Positron](https://github.com/artsy/positron) (our Editorial "Writer" app) and [Force](https://github.com/artsy/force), each boot would come at a high time-cost due to upfront compilation of assets. Tools like [nodemon](https://github.com/remy/nodemon) would automatically stop and start our server process when assets changed but that still didn't alleviate slow iteration times.
```

Comment:
Maybe the following will be clearer for people not familiar with our apps:

Positron (our Publishing CMS called "Writer") and Force (artsy.net)

Reply:
Since I mention Force and Artsy.net in the first sentence I don't think additional context is required here, but I've updated details on Positron!